### PR TITLE
Project MVP

### PR DESCRIPTION
**Pull request by AdnanCodes**

### CODE REVIEW

File: ls8/cpu.py [hunk 1]
```diff
@@ -62,4 +76,26 @@ def trace(self):
...
+            elif self.ir == self.instruction["LDI"]:
+                # Immediate set value of register to an intger
+                self.reg[operand_a] = 8
+                self.pc += 3
+            elif self.ir == self.instruction["PRN"]:
+                print(self.reg[operand_a])
+                self.pc += 2
+            else:
+                print(f"Eroor: Unknown Command {self.ir}")
+                sys.exit(1)
```
Comment (brianilles) [inline ls8/cpu.py:161]: How could we improve this if-else block to be more extensible?

**Review by brianilles — APPROVED** [review]
Great start.

File: ls8/cpu.py (hunk 1 above)
Comment (AdnanCodes) [reply ls8/cpu.py:161]: As I was reading the Step 10, using something call branch which will give better performance python wise

**Review by brianilles — APPROVED** [review]
Excellent work on day 2 tasks

File: ls8/cpu.py [hunk 2]
```diff
@@ -75,30 +91,49 @@ def trace(self):
...
-                self.pc += 2
-            elif self.ir == self.instruction["MUL"]:
-                self.alu("MUL", operand_a, operand_b)
-                self.pc += 3
-            else:
-                print(f"Eroor: Unknown Command {self.ir}")
+                break
+            # try to engage the reads from program and excecute instructions
+            try:
+                self.instruction[self.ir](operand_a, operand_b)
```
Comment (brianilles) [inline ls8/cpu.py:158]: Excellent use of branch-table.

File: ls8/cpu.py [hunk 3]
```diff
@@ -75,30 +91,49 @@ def trace(self):
...
+
+    def handle_MUL(self, operand_a, operand_b):
+        self.alu("MUL", operand_a, operand_b)
+        self.pc += 3
+
+    def handle_POP(self, operand_a, operand_b):
+        value = self.ram[self.reg[SP]]
+        self.reg[operand_a] = value
+        self.reg[SP] += 1
+        self.pc += 2
```
Comment (brianilles) [inline ls8/cpu.py:120]: Nice. Getting the value from memory. Storing the value from the stack in the reg. 👍

**Review by brianilles — APPROVED** [review]

**Pull request merged**